### ISSUE TITLE
Rich text editor corrupts images with UTF-8 SVG data URIs

### ISSUE THREAD

**Issue by personalizedrefrigerator** [laurent22/joplin#8736]
## Environment

Joplin version: 4e25377122c8515b31fe21d575f068d197c86de0
Platform: Desktop
OS specifics: Ubuntu 23.04

## Steps to reproduce

1. Create a new note
2. Open the note in the markdown editor
3. Set the note's content to
```html
<img src="data:image/svg+xml;utf8,<svg width=&quot;1700&quot; height=&quot;1536&quot; xmlns=&quot;http://www.w3.org/2000/svg&quot;> <path d=&quot;m0,0 l100,1000 l200,0 z&quot;/> </svg>">
```
4. Switch to the rich text editor
5. Edit the note
6. Switch back to the markdown editor

## Describe what you expected to happen

The image should be preserved.

## Related

https://github.com/laurent22/joplin/issues/8735 (not loaded resources are represented partially with image tags with UTF-8 SVG data URIs).

**Comment by laurent22** [comment]
I think that's purposely disabled by the sanitizer for security reasons?

**Comment by personalizedrefrigerator** [comment]
The current output after converting from HTML->MD->HTML is
```html
<img src="data:;base64,<svg width=" 1700"="" height="1536" xmlns="http://www.w3.org/2000/svg">
```


Note that it does render correctly in the HTML viewer:

![screenshot of rendered HTML](https://github.com/laurent22/joplin/assets/46334387/aeb77b5d-188a-47f1-8904-865f6a89c874)

**Comment by adrianvg** [comment]
> I think that's purposely disabled by the sanitizer for security reasons?

Seeing this too, but don't understand why this is.
Is the sanitizer I can disable or tweak?

I'm and enduser BTW. :-)

**Comment by personalizedrefrigerator** [comment]
It looks like this is [an upstream bug](https://github.com/tinymce/tinymce/issues/3864) and [was fixed in TinyMCE 6.1](https://github.com/tinymce/tinymce/issues/3864#issuecomment-1545497679). Thus, upgrading TinyMCE should fix this issue.
